### DISCUSSION TITLE
Trailing slash removal in RTMP URL

### DISCUSSION THREAD

**Discussion by vampirefrog** [OvenMediaLabs/OvenMediaEngine#1482] [Help]
Howdy, I noticed that the trailing slash gets removed if I `:startPush` with an RTMP URL that looks like `rtmp://example.com/live/`. This has the result that, if I try to get all the `:pushes` and compare them to what I have in my app's list, the URL won't match, because OME returns `rtmp://example.com/live` without the trailing slash. My solution is to remove the trailing slash when comparing, but is there any other rule that I should be aware of, besides the removal of the trailing slash? So, for example, any sort of normalizing of the URL? Does the 1935 port also get removed? Any other rules besides those? Thanks!

**Comment by vampirefrog** [comment]
Also, what about SRT, does that URL get transformed as well?

**Comment by bchah** [comment]
You are on the right track. Study the documentation, look at the example requests / responses and craft your logic accordingly. Removal of trailing slashes is pretty common.

https://airensoft.gitbook.io/ovenmediaengine/rest-api/v1/virtualhost/application/push
